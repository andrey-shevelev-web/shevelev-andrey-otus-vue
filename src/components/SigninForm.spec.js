import { describe, it, expect } from "vitest";
import { mount, flushPromises } from "@vue/test-utils";
import component from "@/components/SigninForm.vue"

describe('SigninForm', () => {
  it('mounts without error', () => {
    const wrapper = mount(component)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('shows error when login is not valid', async () => {
    const wrapper = mount(component)
    const form = wrapper.find('form')
    const login = wrapper.find('#email1')
    const password = wrapper.find('#password1')
    const button = wrapper.find('[type="submit"]')
    await login.setValue('none@novhere.com')
    await login.trigger('change')
    await login.trigger('keyup', {key: 'enter'})
    await button.trigger('click')
    vi.runAllTimers()
    expect(wrapper.html()).toMatchSnapshot()
  })

})