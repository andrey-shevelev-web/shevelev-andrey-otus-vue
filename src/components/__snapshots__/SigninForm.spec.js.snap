// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SigninForm > mounts without error 1`] = `
"<div class="px-4 py-8 md:px-6 lg:px-8 flex align-items-center justify-content-center">
  <div class="surface-card p-4 shadow-2 border-round w-full lg:w-4">
    <div class="text-center mb-5">
      <div class="text-900 text-3xl font-medium mb-3">DEMO SHOP</div><span class="text-600 font-medium line-height-3">Login to your personal account</span>
    </div>
    <form>
      <div class="mb-3"><label for="email1" class="block text-900 font-medium mb-2">Username</label><input class="p-inputtext p-component w-full mb-1" data-pc-name="inputtext" data-pc-section="root" id="email1" type="text" placeholder="Enter your username"><small class="p-error">&nbsp;</small></div>
      <div class="mb-5"><label for="password1" class="block text-900 font-medium mb-2">Password</label><input class="p-inputtext p-component w-full mb-1" data-pc-name="inputtext" data-pc-section="root" id="password1" type="password" placeholder="Enter your password"><small class="p-error">&nbsp;</small></div>
      <!--v-if--><button class="p-button p-component w-full" type="submit" aria-label="Sign in" data-pc-name="button" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-user" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">Sign in</span>
        <!---->
      </button>
    </form>
  </div>
</div>"
`;

exports[`SigninForm > shows error when login is not valid 1`] = `
"<div class="px-4 py-8 md:px-6 lg:px-8 flex align-items-center justify-content-center">
  <div class="surface-card p-4 shadow-2 border-round w-full lg:w-4">
    <div class="text-center mb-5">
      <div class="text-900 text-3xl font-medium mb-3">DEMO SHOP</div><span class="text-600 font-medium line-height-3">Login to your personal account</span>
    </div>
    <form>
      <div class="mb-3"><label for="email1" class="block text-900 font-medium mb-2">Username</label><input class="p-inputtext p-component p-filled w-full mb-1" data-pc-name="inputtext" data-pc-section="root" id="email1" type="text" placeholder="Enter your username"><small class="p-error">&nbsp;</small></div>
      <div class="mb-5"><label for="password1" class="block text-900 font-medium mb-2">Password</label><input class="p-inputtext p-component w-full mb-1" data-pc-name="inputtext" data-pc-section="root" id="password1" type="password" placeholder="Enter your password"><small class="p-error">&nbsp;</small></div>
      <!--v-if--><button class="p-button p-component w-full" type="submit" aria-label="Sign in" data-pc-name="button" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-user" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">Sign in</span>
        <!---->
      </button>
    </form>
  </div>
</div>"
`;
